docs/source/external_exporters.rst

```diff
@@ -0,0 +1,156 @@
+.. _external_exporter
+
+Using external exporters
+========================
+
+.. versionadded:: 4.2
+
+    You can now use the ``--to`` flag to use custom export formats defined in other
+    libraries.
+
+
+The command-line syntax to run the ``nbconvert`` script is::
+
+  $ jupyter nbconvert --to FORMAT notebook.ipynb
+
+This will convert the Jupyter document file ``notebook.ipynb`` into the output
+format designated by the ``FORMAT`` string as explained below.
+
+A few built-in formats are available by default: `html`, `pdf`,
+`script`, `latex`. Each of these has its own _exporter_ with many configuration
+options that can be extended. Having the option to point to a different _exporter_ 
+allows authors to create their own fully customized templates. 
+
+A custom _exporter_ must be a globally importable python object. We recommend that
+these be distributed as python libraries.
+
+To export using an exporter from an external library, use the `fully qualified`
+name of this exporter on the command line as the argument of the ``--to`` flag::
```

Comment:
I like the idea of entry points, but in the past it seemed like it was too
setuptools centric. That is why the new nbextension stuff didn't use them.

On Wed, Jan 6, 2016 at 9:22 AM, Thomas Kluyver <notifications@github.com>
wrote:

> In docs/source/external_exporters.rst
> <https://github.com/jupyter/nbconvert/pull/141#discussion_r48983354>:
>
> > +
> > +  $ jupyter nbconvert --to FORMAT notebook.ipynb
> > +
> > +This will convert the Jupyter document file ``notebook.ipynb`` into the output
> > +format designated by the ``FORMAT`` string as explained below.
> > +
> > +A few built-in formats are available by default: `html`, `pdf`,
> > +`script`, `latex`. Each of these has its own _exporter_ with many configuration
> > +options that can be extended. Having the option to point to a different _exporter_
> > +allows authors to create their own fully customized templates.
> > +
> > +A custom _exporter_ must be a globally importable python object. We recommend that
> > +these be distributed as python libraries.
> > +
> > +To export using an exporter from an external library, use the `fully qualified`
> > +name of this exporter on the command line as the argument of the ``--to`` flag::
>
> Do we want to do this, or use entry points
> <http://entrypoints.readthedocs.org/en/latest/>, with a group name like
> nbconvert.exporters? I think I'd lean towards entry points, because then
> it's trivial to list all the available exporters you have installed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/nbconvert/pull/141/files#r48983354>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com